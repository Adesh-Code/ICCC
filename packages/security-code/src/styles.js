const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
