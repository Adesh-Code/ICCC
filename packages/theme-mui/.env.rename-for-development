# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE