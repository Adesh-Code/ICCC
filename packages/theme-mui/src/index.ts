// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
