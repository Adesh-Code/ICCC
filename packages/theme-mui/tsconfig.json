{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
