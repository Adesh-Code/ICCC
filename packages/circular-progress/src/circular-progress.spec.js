import React from "react";
import CircularProgress from "./circular-progress";
import { createSerializer, matchers } from "@emotion/jest";
import { screen, render } from "@testing-library/react";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer({ includeStyles: false }));

const arrange = (args) => {
  return render(
    <ThemeProvider theme={light}>
      <CircularProgress {...args} />
    </ThemeProvider>
  );
};

describe("Circular progress", () => {
  it("renders a circular progress", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("renders in medium size", () => {
    arrange({ size: "medium" });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--medium");
  });

  it("renders in large size", () => {
    arrange({ size: "large" });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--large");
  });

  it("renders in large size with label", () => {
    const { container } = arrange({ label: "Label", size: "large" });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--large");
    expect(container).toHaveTextContent("Label");
  });

  it("renders in large size with percentage label", () => {
    const { container } = arrange({
      size: "large",
      variant: "determinate",
      value: 95
    });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--large");
    expect(container).toHaveTextContent("95%");
  });

  it("renders in medium size with percentage label", () => {
    const { container } = arrange({
      size: "medium",
      variant: "determinate",
      value: 64
    });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--medium");
    expect(container).toHaveTextContent("64%");
  });

  it("renders in small size with percentage label", () => {
    const { container } = arrange({
      size: "small",
      variant: "determinate",
      value: 48
    });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--small");
    expect(container).toHaveTextContent("48%");
  });

  it("renders in x-small size with percentage label", () => {
    const { container } = arrange({
      label: "X-small determinate",
      size: "x-small",
      variant: "determinate",
      value: 10
    });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("DhigCircularProgress--size--x-small");
    expect(container).toHaveTextContent("X-small determinate");
  });

  it("accepts classnames", () => {
    arrange({
      className: "foo"
    });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("foo");
  });
});
