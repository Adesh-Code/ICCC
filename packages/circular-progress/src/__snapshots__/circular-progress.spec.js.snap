// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circular progress renders a circular progress 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg DhigCircularProgress--wrapper emotion-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg DhigCircularProgress--container emotion-1"
    >
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root DhigCircularProgress--background DhigCircularProgress--size--small MuiCircularProgress-determinate MuiCircularProgress-colorPrimary emotion-2"
        role="progressbar"
        style="transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg emotion-3"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate emotion-4"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
            style="stroke-dasharray: 126.920; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
      <span
        class="MuiCircularProgress-root DhigCircularProgress--foreground DhigCircularProgress--size--small MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-5"
        role="progressbar"
      >
        <svg
          class="MuiCircularProgress-svg emotion-3"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-7"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
